hyperstone/plugins/loaders/pe.py

```diff
@@ -130,18 +138,53 @@ class PELoader(Plugin):
     Loads PE files from disk.
     interact() can be used to map a PE file.
     """
+
     _INTERACT_TYPE = PELoaderInfo
-    _IAT_HOOK_SEP = '@'
+    _IAT_HOOK_SEP = "@"
+
+    def __init__(self, *files: PELoaderInfo, dll_search_paths: list[str] = None):
+        if dll_search_paths is None:
+            dll_search_paths = []
+
+        self._available_pes: dict[str, str] = {}
+        for directory in reversed(dll_search_paths):
+            current_path = os.path.expandvars(directory)
+            for item in os.listdir(current_path):
+                self._available_pes[item.lower()] = os.path.join(
+                    current_path, item
+                ).lower()
+
+        for item in files:
+            self._available_pes[item.file.lower()] = item.file.lower()
 
-    def __init__(self, *files: PELoaderInfo):
         super().__init__(*files)
+
         self._loaded: Dict[FileNameType, MappedPE] = {}
-        self._phantom_hooks: Dict[str, Dict[str, int]] = {}
+        self._fake_exports: Dict[str, FakeExport] = {}
         self._iat_hooks: Dict[str, Callable[[HyperEmu, dict], Any]] = {}
         self._segment_plugin: Optional[Segment] = None
         self._stream_mapper: Optional[StreamMapper] = None
         self._enforce_plugin: Optional[EnforceMemory] = None
         self._hook_plugin: Optional[Hook] = None
+        self._handled_dependencies: set[str] = []
```

Comment:
`Set[str]`, might need to import it from `typing`
Also, this is a `List`, not a `Set` (which is initiated via `set()`)